### PR TITLE
Enable `-D rustdoc::broken-intra-doc-links`

### PR DESCRIPTION
**Pull request by tcharding**
Make an effort to do some docs improvements without touching the whole codebase.

- Patch 1: Trivial, because I can't help myself.
- Patch 2: Clean up docs on a private method in preparation for duplicating in public docs
- Patch 3: Fix broken links and enable `-D rustdoc::broken-intra-doc-links` in CI

I'm in no way across this codebase, there may be other problems with docs that show up in this PR but the PR is an attempt at improving the docs without going too wild - both for my sanity and that of reviewers.

### CODE REVIEW

File: src/policy/concrete.rs [hunk 1]
```diff
@@ -218,18 +218,18 @@ pub enum PolicyError {
     DuplicatePubKeys,
 }
 
-/// Descriptor context for [`Policy`] compilation into a [`Descriptor`]
+/// Descriptor context for [`Policy`] compilation into a [`crate::Descriptor`].
 pub enum DescriptorCtx<Pk> {
-    /// [Bare][`Descriptor::Bare`]
+    /// See docs for [`crate::Descriptor::Bare`].
```
Comment (tcharding) [inline src/policy/concrete.rs:223]: I'm not super happy with all the `crate::`s here.

**Comment by apoelstra** [comment]
Needs rebase.

File: src/policy/concrete.rs (hunk 1 above)
Comment (tcharding) [reply src/policy/concrete.rs:223]: Does anyone remember how to import a type just for the docs build, we do it someplace else but I can't find it?

**Comment by tcharding** [comment]
I rebased but I'd prefer to import `Descriptor` instead of using `crate::Descriptor` everywhere.

File: src/policy/concrete.rs (hunk 1 above)
Comment (tcharding) [reply src/policy/concrete.rs:212]: Its
```rust
#[cfg(doc)]
use Foo;
```

**Comment by tcharding** [comment]
Used `#[cfg(doc]` imports to improve the docs (remove `crate::` prefix).

**Review by apoelstra — APPROVED** [review]
ACK ebaa31bca7e80f02751e3435907b3aad99d74168

**Comment by tcharding** [comment]
oh, how did that merge? Did you fix the formatting issue?

**Comment by apoelstra** [comment]
What formatting issue?

**Comment by apoelstra** [comment]
Oh, I see, cirrus is gated on rustfmt, and I didn't bother checking it locally. I'll fix it.

**Pull request merged**